### PR TITLE
LCI 20 cards

### PR DESCRIPTION
**Pull request by Simisays**

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/contested_bame_ball.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+Name:Contested Game Ball
+ManaCost:2
+Types:Artifact
+T:Mode$ DamageDoneOnce | CombatDamage$ True | ValidTarget$ You | ValidSource$ Card | TriggerZones$ Battlefield | Execute$ TrigGainGontrol | TriggerDescription$ Whenever you're dealt combat damage, the attacking player gains control of CARDNAME and untaps it.
+SVar:TrigGainGontrol:DB$ GainControl | NewController$ TriggeredAttackingPlayer | Untap$ True
+A:AB$ Draw | Cost$ 2 T | SubAbility$ DBPutCounter | SpellDescription$ Draw a card and put a point counter on CARDNAME. Then if it has five or more point counters on it, sacrifice it and create a Treasure token.
+SVar:DBPutCounter:DB$ PutCounter | CounterType$ POINT | CounterNum$ 1 | SubAbility$ DBSac
+SVar:DBSac:DB$ Sacrifice | SacValid$ Self | ConditionPresent$ Card.Self+counters_GE5_POINT | RememberSacrificed$ True |  SubAbility$ DBTreasure
+SVar:DBTreasure:DB$ Token | TokenScript$ c_a_treasure_sac | ConditionDefined$ Remembered | ConditionPresent$ Card | SubAbility$ DBCleanup
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/contested_bame_ball.txt:9]: better use Branch

File: forge-gui/res/cardsfolder/upcoming/corpses_of_the_lost.txt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+Name:Corpses of the Lost
+ManaCost:2 B
+Types:Enchantment
+S:Mode$ Continuous | Affected$ Creature.Skeleton+YouCtrl | AddPower$ 1 | AddKeyword$ Haste | Description$ Skeletons you control get +1/+0 and have haste.
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigToken | TriggerDescription$ When CARDNAME enters the battlefield, create a 2/2 black Skeleton Pirate creature token.
+SVar:TrigToken:DB$ Token | TokenScript$ b_2_2_skeleton_pirate
+T:Mode$ Phase | Phase$ End of Turn | ValidPlayer$ You.descended | TriggerZones$ Battlefield | Execute$ TrigReturn | TriggerDescription$ At the beginning of your end step, if you descended this turn, you may pay 1 life. If you do, return Corpses of the Lost to its owner’s hand. (You descended if a permanent card was put into your graveyard from anywhere.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/corpses_of_the_lost.txt:7]: CARDNAME

File: forge-gui/res/cardsfolder/upcoming/deathcap_marionette.txt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+Name:Deathcap Marionette
+ManaCost:1 B
+Types:Creature Fungus
+PT:1/1
+K:Deathtouch 
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | OptionalDecider$ True | ValidCard$ Card.Self | Execute$ TrigMill | TriggerDescription$ When CARDNAME enters the battlefield, you may mill two cards. (You may put the top two cards of your library into your graveyard.)
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/deathcap_marionette.txt:6]: move optional to effect

File: forge-gui/res/cardsfolder/upcoming/deconstruction_hammer.txt [hunk 4]
```diff
@@ -0,0 +1,8 @@
+Name:Deconstruction Hammer
+ManaCost:W
+Types:Artifact Equipment
+DeckHas:Ability$Sacrifice
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/deconstruction_hammer.txt:4]: move lower

File: forge-gui/res/cardsfolder/upcoming/deconstruction_hammer.txt [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+S:Mode$ Continuous | Affected$ Creature.EquippedBy | AddPower$ 1 | AddToughness$ 1 | AddAbility$ ABDestroy | Description$ Equipped creature gets +1/+1 and has "{3}, {T}, Sacrifice Deconstruction Hammer: Destroy target artifact or enchantment."
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/deconstruction_hammer.txt:5]: CARDNAME

File: forge-gui/res/cardsfolder/upcoming/deathcap_marionette.txt [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+SVar:TrigMill:DB$ Mill | Defined$ You | NumCards$ 2
```
Comment (Hanmac) [inline forge-gui/res/cardsfolder/upcoming/deathcap_marionette.txt:7]: @tool4ever @Northmoc 
the "you may mill two" , i don't know if that is a rules change,
but the release notes say this:

> If you don't have at least two cards in your library, you can't choose to mill two cards.

meaning we can't use Optional Trigger there?
or does Mill Effect can handle this?

for example, `Crawling Sensation` doesn't have the ruling yet?

File: forge-gui/res/cardsfolder/upcoming/deepfathom_echo.txt [hunk 7]
```diff
@@ -0,0 +1,9 @@
+Name:Deepfathom Echo
+ManaCost:2 G U
+Types:Creature Merfolk Spirit
+PT:4/4
+T:Mode$ Phase | Phase$ BeginCombat | ValidPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigExplore | TriggerDescription$ At the beginning of combat on your turn, CARDNAME explores. Then you may have it become a copy of another creature you control until end of turn. (To have this creature explore, reveal the top card of your library. Put that card into your hand if it's a land. Otherwise, put a +1/+1 counter on this creature, then put the card back or put it into your graveyard.)
+SVar:TrigExplore:DB$ Explore | SubAbility$ DBClone
+SVar:DBClone:DB$ Clone | ValidTgts$ Creature.Other+YouCtrl | TgtPrompt$ Select another creature you control | Optional$ True | AILogic$ CloneBestCreature | Duration$ UntilEndOfTurn
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/deepfathom_echo.txt:7]: doesn't target

File: forge-gui/res/cardsfolder/upcoming/dreadmaws_ire.txt [hunk 8]
```diff
@@ -0,0 +1,8 @@
+Name:Dreadmaw's Ire
+ManaCost:R
+Types:Instant
+A:SP$ Pump | ValidTgts$ Creature.attacking | TgtPrompt$ Select target attacking creature | NumAtt$ +2 | NumDef$ +2 | KW$ Trample | SubAbility$ DBAnimate | SpellDescription$ Until end of turn, target attacking creature gets +2/+2 and gains trample and "Whdeals combat damage to a player, destroy target artifact that player controls."
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/dreadmaws_ire.txt:4]: desc typo

File: forge-gui/res/cardsfolder/upcoming/earthshaker_dreadmaw.txt [hunk 9]
```diff
@@ -0,0 +1,10 @@
+Name:Earthshaker Dreadmaw
+ManaCost:4 G G
+Types:Creature Dinosaur
+PT:6/6
+K:Trample
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigDraw | TriggerDescription$ When CARDNAME enters the battlefield, draw a card for each other Dinosaur you control.
+SVar:TrigDraw:DB$ Draw | Defined$ You | NumCards$ X
+SVar:X:Count$Valid Dinosaur.Other+YouCtrl
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/earthshaker_dreadmaw.txt:8]: StrictlyOther

File: forge-gui/res/cardsfolder/upcoming/deathcap_marionette.txt (hunk 6 above)
Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/deathcap_marionette.txt:7]: it's been in the CR for a while and we support it since 62bc1b4

Comment (Hanmac) [reply forge-gui/res/cardsfolder/upcoming/deathcap_marionette.txt:7]: ah, so they just didn't update the ruling on the cards

File: forge-gui/res/cardsfolder/upcoming/contested_bame_ball.txt (hunk 1 above)
Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/contested_bame_ball.txt:9]: Hmm can you do a branch when there's only 1 possible result (nothing should happen when it has 4 or less point counters) ?

Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/contested_bame_ball.txt:9]: sure

Comment (Hanmac) [reply forge-gui/res/cardsfolder/upcoming/contested_bame_ball.txt:9]: checkout treasure map:
https://github.com/Card-Forge/forge/blob/e1ea01f06584a9b42c4422387b6735b979b33ceb/forge-gui/res/cardsfolder/t/treasure_map_treasure_cove.txt#L6-L9

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/contested_bame_ball.txt:9]: Thanks will do

Comment (Hanmac) [reply forge-gui/res/cardsfolder/upcoming/contested_bame_ball.txt:9]: Also for the Ball, doesn't say "if you do" for the Sacrifice.

so even if you can't sacrifice it, it should still make the Token.

so no RememberSacrificed needed

File: forge-gui/res/cardsfolder/upcoming/explorers_cache.txt [hunk 10]
```diff
@@ -0,0 +1,10 @@
+Name:Explorer's Cache
+ManaCost:1 G
+Types:Artifact
+K:etbCounter:P1P1:2
+T:Mode$ ChangesZone | Origin$ Battlefield | Destination$ Graveyard | ValidCard$ Creature.YouCtrl+counters_GE1_P1P1 | TriggerZones$ Battlefield | Execute$ TrigPutCounter | TriggerDescription$ Whenever a creature you control with a +1/+1 counter on it dies, put a +1/+1 counter on CARDNAME.
+SVar:TrigPutCounter:DB$ PutCounter | Defined$ Self | CounterType$ P1P1 | CounterNum$ 1
+A:AB$ MoveCounter | Cost$ T | ValidTgts$ Creature | Source$ Self | CounterType$ P1P1 | CounterNum$ 1 | SorcerySpeed$ True | SpellDescription$ Move a +1/+1 counter from Explorer’s Cache onto target creature. Activate only as a sorcery.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/explorers_cache.txt:7]: CARDNAME

**Review by tool4ever — APPROVED** [review]

**Pull request merged**